### PR TITLE
[Fix for #1929] HeaderModelBinder needs to honor explicit name.

### PR DESCRIPTION
**Pull request by harshgMSFT**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.ModelBinding/Binders/HeaderModelBinder.cs [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
 using System.Reflection;
 using System.Threading.Tasks;
 using Microsoft.AspNet.Mvc.ModelBinding.Internal;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
+    /// <summary>
+    /// A <see cref="MetadataAwareBinder{IHeaderBinderMetadata}"/> which uses <see cref="Http.HttpRequest.Headers"/>
+    /// to bind the model.
+    /// </summary>
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ModelBinding/Binders/HeaderModelBinder.cs]: thanks for doing the cleanup here :heart_eyes:

File: test/Microsoft.AspNet.Mvc.ModelBinding.Test/Binders/HeaderModelBinderTests.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            // Arrange
+            var type = typeof(string[]);
+            var header = "Accept";
+            var headerValue = "application/json,text/json";
+            var binder = new HeaderModelBinder();
+            var modelBindingContext = GetBindingContext(type);
+
+            modelBindingContext.ModelName = header;
+            modelBindingContext.OperationBindingContext.HttpContext.Request.Headers.Add(header, new[] { headerValue });
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.ModelBinding.Test/Binders/HeaderModelBinderTests.cs]: so.much.property.access.chaining

**Comment by rynowak** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
Add a test that verifies we get modelstate errors when header value is missing

**Comment by yishaigalatzer** [comment]
:watch:

**Comment by harshgMSFT** [comment]
Well we get model state error based on data annotations... From header is not related...if there is no value from header we do not add a model state error unless there is a required attribute. Curious if you had some specific scenario here.

**Comment by yishaigalatzer** [comment]
Verify this specifically even if it is just for a sample (header + required)


-----Original Message-----

**Comment by yishaigalatzer** [comment]
:Shipit: when test there

**Comment by rynowak** [comment]
@harshgMSFT is this still open?

**Comment by harshgMSFT** [comment]
nope e805e67b4c96dd619daab4ff57f5bea2d0b5c3f9

**Pull request closed without merging**